lms/lib/courseware_search/test/test_lms_filter_generator.py

```diff
@@ -66,7 +68,53 @@ def test_user_not_provided(self):
         """
         Tests that we get empty list of courses in case the user is not provided
         """
-        field_dictionary, filter_dictionary = LmsSearchFilterGenerator.generate_field_filters()
+        field_dictionary, filter_dictionary, _ = LmsSearchFilterGenerator.generate_field_filters()
 
         self.assertTrue('start_date' in filter_dictionary)
         self.assertEqual(0, len(field_dictionary['course']))
+
+    def test_excludes_microsite(self):
+        """ By default there is the test microsite to exclude """
+        _, _, exclude_dictionary = LmsSearchFilterGenerator.generate_field_filters(user=self.user)
+        self.assertIn('org', exclude_dictionary)
+        exclude_orgs = exclude_dictionary['org']
+        self.assertEqual(1, len(exclude_orgs))
+        self.assertEqual('TestMicrositeX', exclude_orgs[0])
+
+    @patch('microsite_configuration.microsite.get_all_orgs', Mock(return_value=[]))
+    def test_excludes_no_microsite(self):
+        """ Test when no microsite is present - nothing to exclude """
+        _, _, exclude_dictionary = LmsSearchFilterGenerator.generate_field_filters(user=self.user)
+        self.assertNotIn('org', exclude_dictionary)
+
+    @patch('microsite_configuration.microsite.get_value', Mock(return_value='TestMicrositeX'))
+    def test_excludes_microsite_within(self):
+        field_dictionary, _, exclude_dictionary = LmsSearchFilterGenerator.generate_field_filters(user=self.user)
+        self.assertNotIn('org', exclude_dictionary)
+        self.assertIn('org', field_dictionary)
+        self.assertEqual('TestMicrositeX', field_dictionary['org'])
+
+    @patch(
+        'microsite_configuration.microsite.get_all_orgs',
+        Mock(return_value=["TestMicrosite1", "TestMicrosite2", "TestMicrosite3", "TestMicrosite4"])
+    )
+    def test_excludes_multi_microsites(self):
+        _, _, exclude_dictionary = LmsSearchFilterGenerator.generate_field_filters(user=self.user)
+        self.assertIn('org', exclude_dictionary)
+        exclude_orgs = exclude_dictionary['org']
+        self.assertEqual(4, len(exclude_orgs))
+        self.assertIn('TestMicrosite1', exclude_orgs)
+        self.assertIn('TestMicrosite2', exclude_orgs)
+        self.assertIn('TestMicrosite3', exclude_orgs)
+        self.assertIn('TestMicrosite4', exclude_orgs)
+
+    @patch(
+        'microsite_configuration.microsite.get_all_orgs',
+        Mock(return_value=["TestMicrosite1", "TestMicrosite2", "TestMicrosite3", "TestMicrosite4"])
+    )
+    @patch('microsite_configuration.microsite.get_value', Mock(return_value='TestMicrosite3'))
+    def test_excludes_multi_microsites_within(self):
+        field_dictionary, _, exclude_dictionary = LmsSearchFilterGenerator.generate_field_filters(user=self.user)
+        self.assertNotIn('org', exclude_dictionary)
+        self.assertIn('org', field_dictionary)
+        self.assertEqual('TestMicrosite3', field_dictionary['org'])
```

Comment:
Yes, mock service can support such a test - but having said that, they would be essentially duplicating the tests from within edx-search... but happy to add them nevertheless.